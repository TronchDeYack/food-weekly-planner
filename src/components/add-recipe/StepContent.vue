<template>
  <v-stepper-content :step="step">
    <slot></slot>
    <StepActions :step="step" @next-step="nextStep" @previous-step="previousStep"/>
  </v-stepper-content>
</template>

<script>
import StepActions from '@/components/add-recipe/StepActions.vue';

export default {
  name: 'add-recipe-step-content',
  components: {
    StepActions,
  },
  props: {
    step: {
      type: Number,
    },
  },
  // TODO: remove that by using a store
  methods: {
    nextStep(event) {
      this.$emit('next-step', event);
    },
    previousStep(event) {
      this.$emit('previous-step', event);
    },
  },
};
</script>
