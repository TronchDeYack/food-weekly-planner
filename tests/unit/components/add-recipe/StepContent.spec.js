import StepContent from '@/components/add-recipe/StepContent.vue';

import { createComponent } from '../../../utils';

describe('StepContent', () => {
  const propsData = {
    step: 2,
  };

  describe('render', () => {
    it('Should render.', () => {
      const wrapper = createComponent(StepContent, propsData);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
