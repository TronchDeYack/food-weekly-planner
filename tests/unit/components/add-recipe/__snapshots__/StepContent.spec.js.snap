// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepContent render Should render. 1`] = `
<v-stepper-content-stub step="2">
  <stepactions-stub step="2"></stepactions-stub>
</v-stepper-content-stub>
`;
